### DISCUSSION TITLE
in mongodb , allow _id after serialization

### DISCUSSION THREAD

**Discussion by rubiin** [mikro-orm/mikro-orm#5964] [General]
using mikroorm

``` ts 

//base entity

export abstract class BaseEntity {
  @PrimaryKey({ index: true })
  _id: ObjectId;

  @Property()
  deletedAt?: Date;

  @Property()
  createdAt = new Date();


}
```

```ts

@Entity()
export class Answer extends BaseEntity {
  @Property({ length: 100_000 })
  answer!: string;
}
```

Repro: [Link](https://codesandbox.io/p/devbox/jolly-hopper-vfrn3k?workspaceId=34d9c2b6-158a-438c-8bc3-6590c0d0323a&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm05105ts00052v6ibccwpd4q%2522%252C%2522sizes%2522%253A%255B52.79181708784597%252C47.20818291215403%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm05105tr00022v6i6cgtg9iq%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm05105tr00032v6in54m6vgj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm05105tr00042v6i3wvqmvpq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm05105tr00022v6i6cgtg9iq%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm05105tr00012v6i6ev5n7ge%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cm0511t0z00022v6j9gvl13hc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A17%252C%2522startColumn%2522%253A24%252C%2522endLineNumber%2522%253A17%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fexample.test.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cm05105tr00022v6i6cgtg9iq%2522%252C%2522activeTabId%2522%253A%2522cm0511t0z00022v6j9gvl13hc%2522%257D%252C%2522cm05105tr00042v6i3wvqmvpq%2522%253A%257B%2522id%2522%253A%2522cm05105tr00042v6i3wvqmvpq%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522cm05105tr00032v6in54m6vgj%2522%253A%257B%2522id%2522%253A%2522cm05105tr00032v6in54m6vgj%2522%252C%2522activeTabId%2522%253A%2522cm0513vy400252v6j02qkyda2%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522cm0511zmt00122v6j64kfc49n%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522npm%2520test%2522%252C%2522id%2522%253A%2522cm0511zou001f2v6j3jx77ql6%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522cm0513vy400252v6j02qkyda2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cm0513w17001cddj31hpi8wyl%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by rubiin** [comment]
Two different results when serializing and on console. Is there a way to make `_id` available on both cases. This is to maintain consitency as using the aggregate method doesnt create id field 

![image](https://github.com/user-attachments/assets/8e7e0e1d-f79e-4c28-988a-2980d2648d9d)

**Reply by B4nan** [reply]
If you don't like the `id` property, just dont use it. You could also define it as a get/set yourself instead of using the `SerializedPrimaryKey` decorator, just to have the string value at hand.

**Reply by rubiin** [reply]
Never used anything expect mongoose, and it does return `_id` in its find until I fell in love with mikroorm.

I tried this 

```ts

export abstract class BaseEntity {
  @PrimaryKey({ fieldName: "_id" })
  id: ObjectId;

  @SerializedPrimaryKey()
  _id: string;
}

```
saw an error, maybe `SerializedPrimaryKey` always expects _id to be a primarykey  and  to be defined

![image](https://github.com/user-attachments/assets/e3acae3f-212d-46e7-a296-031eec37d8ec)
